internal/testutil/acc/shared_resource.go

```diff
@@ -87,10 +85,23 @@ func ClusterNameExecution(tb testing.TB) (projectID, clusterName string) {
 	return sharedInfo.projectID, sharedInfo.clusterName
 }
 
+// SerialSleep waits a few seconds so clusters are not created concurrently in the execution project, see HELP-65223.
+// This must be called once the test is marked as parallel, e.g. in PreCheck inside Terraform tests.
+func SerialSleep(tb testing.TB) {
+	tb.Helper()
+	SkipInUnitTest(tb)
+	require.True(tb, sharedInfo.init, "SetupSharedResources must called from TestMain test package")
+
+	sharedInfo.muSleep.Lock()
+	defer sharedInfo.muSleep.Unlock()
+	time.Sleep(10 * time.Second)
```

Comment:
This means parallel test N will be delayed by 10s x N? Why 10s?

Reply:
correct. 10s to pick a number that is big enough to make sure clusters are not running very close. we're not still sure if this is the issue.

 if we validate this is the problem, then we can get the number lower

Reply:
Okay. Thanks for explaining and investigating this 👍